source/dofs/dof_tools_constraints.cc

```diff
@@ -1723,13 +1723,49 @@ namespace DoFTools
   }
 
 
+  namespace internal
+  {
+    template <int dim, int spacedim>
+    void check_if_fe_okay(const FiniteElement<dim,spacedim> &fe)
+    {
+      Assert(fe.get_name().find("FE_Q_Hierarchical<") == std::string::npos,
+             ExcNotImplemented("Constraints for FE_Q_Hierarchical are currently not implemented!"));
```

Comment:
We should use a more descriptive name, e.g., `check_if_fe_supports_constraints`.